drivers/backup/utils/backup_utils.go

```diff
@@ -0,0 +1,292 @@
+func ProcessError(err error, debugMessage ...string) error {
+	if err == nil {
+		return nil
+	}
+	_, file, line, _ := runtime.Caller(1)
+	file = strings.TrimPrefix(file, GlobalTorpedoWorkDirectory)
+	callerInfo := fmt.Sprintf("%s:%d", file, line)
+	debugInfo := "no debug message"
+	if len(debugMessage) > 0 {
+		debugInfo = "debug message: " + debugMessage[0]
+	}
+	processedError := fmt.Errorf("%s\n  at %s <-> %s", err.Error(), callerInfo, debugInfo)
+	return processedError
+}
+
+// GetKubeconfigKeysFromEnv returns kubeconfigKeys from the environment variable KUBECONFIGS
+func GetKubeconfigKeysFromEnv() []string {
+	envVarKubeconfigs := os.Getenv("KUBECONFIGS")
+	kubeconfigKeys := strings.Split(envVarKubeconfigs, ",")
+	return kubeconfigKeys
+}
+
+// GetClusterConfigPath saves the retrieved kubeconfig from the config-map and returns the file path
+func GetClusterConfigPath(kubeconfigKey string, configMapName string, configMapNamespace string) (string, error) {
+	filePath := fmt.Sprintf("%s/%s", GlobalKubeconfigDirectory, kubeconfigKey)
+	if _, err := os.Stat(filePath); err == nil {
+		return filePath, nil
+	}
+	cm, err := core.Instance().GetConfigMap(configMapName, configMapNamespace)
+	if err != nil {
+		debugMessage := fmt.Sprintf("config map: name [%s], namespace [%s]", configMapName, configMapNamespace)
+		return "", ProcessError(err, debugMessage)
+	}
+	kubeconfig, ok := cm.Data[kubeconfigKey]
+	if !ok {
+		err = fmt.Errorf("kubeconfig key [%s] not found in the config map [%s]", kubeconfigKey, configMapName)
+		return "", ProcessError(err)
+	}
+	err = os.WriteFile(filePath, []byte(kubeconfig), 0644)
+	if err != nil {
+		debugMessage := fmt.Sprintf("kubeconfig file: path [%s]; kubeconfig [%s]", filePath, kubeconfig)
+		return "", ProcessError(err, debugMessage)
+	}
+	return filePath, nil
+}
+
+// GetSourceClusterConfigPath returns the file path of the source cluster kubeconfig
+func GetSourceClusterConfigPath() (string, error) {
+	kubeconfigKeys := GetKubeconfigKeysFromEnv()
+	if len(kubeconfigKeys) < 2 {
+		err := fmt.Errorf("the environment variable KUBECONFIGS should have two kubeconfig keys: one for the source cluster and one for the destination cluster")
+		debugMessage := fmt.Sprintf("kubeconfig-keys: [%s]", kubeconfigKeys)
+		return "", ProcessError(err, debugMessage)
+	}
+	sourceClusterKubeconfigKey := kubeconfigKeys[0]
+	sourceClusterConfigPath, err := GetClusterConfigPath(sourceClusterKubeconfigKey, DefaultConfigMapName, DefaultConfigMapNamespace)
+	if err != nil {
+		debugMessage := fmt.Sprintf("source-cluster: kubeconfig-key [%s]", sourceClusterKubeconfigKey)
+		return "", ProcessError(err, debugMessage)
+	}
+	return sourceClusterConfigPath, nil
+}
+
+// GetDestinationClusterConfigPath returns the file path of the destination cluster kubeconfig
+func GetDestinationClusterConfigPath() (string, error) {
+	kubeconfigKeys := GetKubeconfigKeysFromEnv()
+	if len(kubeconfigKeys) < 2 {
```

Comment:
we can have more than 2 clusters in the config , so we remove hardcoding and have different way of fetching and validation , can fetch based on keys